custom/icds_reports/static/js/directives/download/download.directive.js

```diff
@@ -385,8 +387,21 @@ function DownloadController($rootScope, $location, locationHierarchy, locationsS
             vm.onSelectYear({'id': vm.selectedYear});
             vm.selectedFormat = 'xlsx';
         }
+
+        vm.adjustSelectedLevelForNoViewByFilter();
+
     };
 
+    /**
+     * To Adjust selectedLevel for the reports does not have viewBy filters
+     * end up having aggregation_leve set by viewBy filter in the last report selected.
```

Comment:
I read this 3 times and I'm still not sure i understand the intention of it. Can you clean up / rephrase and fix typos?

Reply:
hi cory, What is happening in the UI of tabular report is that, When you select as report example child report  and select and location filter say you selected a state and in the view by filter you selected view by 'awc' which makes the selectedLeve(aggregation_level param in payload) as 5. Which is correct for that report. But after that if you go to other report which does not have that viewby filter like THR report. and directly click on export without touching the location filter. The aggregation_level(selectedLevel) goes as 5 in the payload even though you have state selected in the location filter. This is to fix that, because in this case aggregateion_level=1 is to be sent.